lib/timers.js

```diff
@@ -13,13 +13,15 @@ function onTimeout () {
   while (idx < len) {
     const timer = fastTimers[idx]
 
-    if (timer.expires && fastNow >= timer.expires) {
-      timer.expires = 0
+    if (timer.state === 0) {
+      timer.state = fastNow + timer.delay
+    } else if (timer.state > 0 && fastNow >= timer.state) {
```

Comment:
@ronag 
expired check is executed in next event loop(after 1000msec).
it is executed in same event loop?
```suggestion
    }
    if (timer.state > 0 && fastNow >= timer.state) {
```